nucypher/config/base.py

```diff
@@ -0,0 +1,139 @@
+        """
+        Return a dictionary of JSON serializable configuration key/value pairs
+        matching the input specification of this classes __init__.
+
+        Recommended subclass implementations:
+
+        ```
+        payload = dict(**super().static_payload(), key=value)
+        return payload
+        ```
+
+        OR
+
+        ```
+        subclass_payload = {'key': 'value'}
+        payload = {**super().static_payload(), **subclass_payload}
+        return payload
+        ```
+
+        """
+        payload = dict(config_root=self.config_root)
+        return payload
+
+    @classmethod
+    def generate_filename(cls, modifier: str = None) -> str:
+        name = cls._NAME.lower()
+        if modifier:
+            name += f'-{modifier}'
+        filename = f'{name}.{cls._CONFIG_FILE_EXTENSION.lower()}'
+        return filename
+
+    @classmethod
+    def default_filepath(cls):
+        filename = cls.generate_filename()
+        default_path = os.path.join(cls.DEFAULT_CONFIG_ROOT, filename)
+        return default_path
+
+    def generate_filepath(self, filepath: str = None, modifier: str = None, override: bool = False) -> str:
+        if not filepath:
+            filename = self.generate_filename()
+            filepath = os.path.join(self.config_root, filename)
+        if os.path.exists(filepath) and not override:
+            if not modifier:
+                raise FileExistsError(f"{filepath} exists and no filename modifier supplied.")
+            filename = self.generate_filename(modifier=modifier)
+            filepath = os.path.join(self.config_root, filename)
+        self.filepath = filepath
+        return filepath
+
+    def to_configuration_file(self, filepath: str = None, modifier: str = None, override: bool = False) -> str:
+        filepath = self.generate_filepath(filepath=filepath, modifier=modifier, override=override)
+        filepath = self._write_configuration_file(filepath=filepath, override=override)
+        return filepath
+
+    @classmethod
+    def from_configuration_file(cls, filepath: str = None, **overrides) -> 'BaseConfiguration':
+        filepath = filepath or cls.default_filepath()
+        payload = cls._read_configuration_file(filepath=filepath)
+        instance = cls(filepath=filepath, **payload, **overrides)
+        return instance
+
+    @classmethod
+    def _read_configuration_file(cls, filepath: str) -> dict:
+        try:
+            with open(filepath, 'r') as file:
+                raw_contents = file.read()
+                payload = cls.deserialize(raw_contents)
+        except FileNotFoundError:
+            raise
+        return payload
+
+    def _write_configuration_file(self, filepath: str, override: bool = False) -> str:
+        if os.path.exists(filepath) and not override:
+            raise FileExistsError
+        try:
+            with open(filepath, 'w') as file:
+                file.write(self.serialize())
+        except FileNotFoundError:
```

Comment:
What's the point of this try block? If `FileNotFoundError` happens - we raise it without modification, if some other error happens - we also raise it. Did you want some extra functionality there?
Same question for `serialize` and `deserialize`

Reply:
Nothing (almost). At the moment this does not change the functionality at all, and `raise` accurately. I tend to wrap new code in the exception I think it can throw as a place for a breakpoint, or as a stub for changes as the new API gets used a bit more heavily.  I can add `TODO`s or just remove them.

Reply:
Ahh, I see

Reply:
removed them. :heavy_check_mark: